cmd/audiusd/main.go

```diff
@@ -356,19 +352,144 @@ func connectGET[Req any, Res any](
 			return c.JSON(http.StatusInternalServerError, map[string]string{"error": err.Error()})
 		}
 
-		msg, ok := any(resp.Msg).(proto.Message)
+		respMsg, ok := any(resp.Msg).(proto.Message)
 		if !ok {
 			return c.JSON(http.StatusInternalServerError, map[string]string{"error": "response is not proto.Message"})
 		}
 
-		jsonBytes, err := marshalOpts.Marshal(msg)
+		jsonBytes, err := marshalOpts.Marshal(respMsg)
 		if err != nil {
 			return c.JSON(http.StatusInternalServerError, map[string]string{"error": fmt.Sprintf("failed to marshal response: %v", err.Error())})
 		}
 
 		return c.JSONBlob(http.StatusOK, jsonBytes)
+	}
+}
 
+// setProtobufField sets a protobuf field value based on query parameter values
+func setProtobufField(msgReflect protoreflect.Message, field protoreflect.FieldDescriptor, values []string) error {
```

Comment:
this is not something we get for free from some package? seems strange that we'd have to implement this ourselves